### PR TITLE
Refactor buildslave related code

### PR DESCRIPTION
**Pull request by sa2ajj**
* the most important thing is moving handling of commong parameters to base classes

### CODE REVIEW

**Comment by tardyp** [comment]
ahah this is good merge conflicts with my work :)

**Comment by sa2ajj** [comment]
I think we need to simplify it then improve. ;)

**Comment by tardyp** [comment]
Fully agreed

Le mar. 24 mars 2015 21:51, Mikhail Sobolev <notifications@github.com> a
écrit :

File: master/buildbot/buildslave/ec2.py [hunk 1]
```diff
@@ -287,6 +279,13 @@ def start_instance(self, build):
         else:
             return threads.deferToThread(self._start_instance)
 
+    def _failed_to_start(self):
+        log.msg('%s %s failed to start instance %s (%s)' %
+                (self.__class__.__name__, self.slavename,
+                    self.instance.id, self.instance.state))
+        raise LatentBuildSlaveFailedToSubstantiate(
+            self.instance.id, self.instance.state)
+
```
Comment (seankelly) [inline master/buildbot/buildslave/ec2.py]: Might be useful to put this in the `AbstractLatentBuildSlave` class. `OpenStackLatentBuildSlave` has a similar line for reporting failure to start.

File: master/buildbot/buildslave/ec2.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 from twisted.python import log
 
 from buildbot import config
-from buildbot import interfaces
+from buildbot.interfaces import LatentBuildSlaveFailedToSubstantiate
```
Comment (seankelly) [inline master/buildbot/buildslave/ec2.py]: Could apply the same change to `docker.py` and `openstack.py`.

File: master/buildbot/buildslave/ec2.py (hunk 1 above)
Comment (sa2ajj) [inline master/buildbot/buildslave/ec2.py]: The `instance` member is not part of the abstract class...

I'd appreciate a suggestion on how to deal with it :)

File: master/buildbot/buildslave/ec2.py (hunk 2 above)
Comment (sa2ajj) [inline master/buildbot/buildslave/ec2.py]: done

File: master/buildbot/buildslave/libvirt.py [hunk 3]
```diff
@@ -195,7 +186,7 @@ def _find_existing_instance(self):
         domains = yield self.connection.all()
         for d in domains:
             name = yield d.name()
-            if name.startswith(self.name):
+            if name.startswith(self.slavename):
```
Comment (tardyp) [inline master/buildbot/buildslave/libvirt.py]: I would have to undo this, if we are deprecating slavename..

File: master/buildbot/buildslave/ec2.py (hunk 1 above)
Comment (seankelly) [inline master/buildbot/buildslave/ec2.py]: Could make `_failed_to_start` take two parameters: the instance's id and state.

Comment (sa2ajj) [inline master/buildbot/buildslave/ec2.py]: Sounds good.  I'll see how it goes then.

File: master/buildbot/buildslave/libvirt.py (hunk 3 above)
Comment (sa2ajj) [inline master/buildbot/buildslave/libvirt.py]: I'm inclined toward `name`.  However, `.name` was not available.  It probably needs to be fixed up in the chain...

Comment (tardyp) [inline master/buildbot/buildslave/libvirt.py]: What do you mean? It was there before your patch.

Le jeu. 26 mars 2015 06:56, Mikhail Sobolev <notifications@github.com> a
écrit :

Comment (sa2ajj) [inline master/buildbot/buildslave/libvirt.py]: I mean that without this particular change things fail.  Which leads me to think that `.name` is not available.

Comment (tardyp) [inline master/buildbot/buildslave/libvirt.py]: I think slavename is needed everywhere in process. that's why my version is aliasing slavename to name via a @property

**Comment by sa2ajj** [comment]
How do we proceed with this PR?

**Comment by tardyp** [comment]
:+1: shipit!

**Pull request merged**